ccl/schemachangerccl: TestSubzonesRemovedByGCAfterIndexSwap failed [retry timeout]

ccl/schemachangerccl.TestSubzonesRemovedByGCAfterIndexSwap [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19576651?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19576651?buildTab=artifacts#/) on master @ [df7d3153243b3b680cdc2c7b8474bbd6f11ed9e9](https://github.com/cockroachdb/cockroach/commits/df7d3153243b3b680cdc2c7b8474bbd6f11ed9e9):


```
        	            	  | 3, old_au, 5, /3/"AU", /3/"AU"/9131
        	            	  | 3, old_au, 5, /3/"NZ", /3/"NZ"/9131
        	            	  | 3, yung_au, 6, /3/"AU"/9131, /3/"AU"/PrefixEnd
        	            	  | 3, yung_au, 6, /3/"NZ"/9131, /3/"NZ"/PrefixEnd
        	            	  | 3, NULL, 7, /3, /3/"AU"
        	            	  | 3, NULL, 7, /3/"AU"/PrefixEnd, /3/"CA"
        	            	  | 3, NULL, 7, /3/"CA"/PrefixEnd, /3/"NZ"
        	            	  | 3, NULL, 7, /3/"NZ"/PrefixEnd, /3/"US"
        	            	  | 3, NULL, 7, /3/"US"/PrefixEnd, /4
        	            	  | 2, NULL, 8, /2, NULL
        	            	  |
        	            	  | got:
        	            	  | 1, north_america, 4, /1/"CA", NULL
        	            	  | 1, north_america, 4, /1/"US", NULL
        	            	  | 3, north_america, 4, /3/"CA", NULL
        	            	  | 3, north_america, 4, /3/"US", NULL
        	            	  | 4, north_america, 4, /4/"CA", NULL
        	            	  | 4, north_america, 4, /4/"US", NULL
        	            	  | 1, old_au, 5, /1/"AU", /1/"AU"/9131
        	            	  | 1, old_au, 5, /1/"NZ", /1/"NZ"/9131
        	            	  | 3, old_au, 5, /3/"AU", /3/"AU"/9131
        	            	  | 3, old_au, 5, /3/"NZ", /3/"NZ"/9131
        	            	  | 4, old_au, 5, /4/"AU", /4/"AU"/9131
        	            	  | 4, old_au, 5, /4/"NZ", /4/"NZ"/9131
        	            	  | 1, yung_au, 6, /1/"AU"/9131, /1/"AU"/PrefixEnd
        	            	  | 1, yung_au, 6, /1/"NZ"/9131, /1/"NZ"/PrefixEnd
        	            	  | 3, yung_au, 6, /3/"AU"/9131, /3/"AU"/PrefixEnd
        	            	  | 3, yung_au, 6, /3/"NZ"/9131, /3/"NZ"/PrefixEnd
        	            	  | 4, yung_au, 6, /4/"AU"/9131, /4/"AU"/PrefixEnd
        	            	  | 4, yung_au, 6, /4/"NZ"/9131, /4/"NZ"/PrefixEnd
        	            	  | 1, NULL, 7, /1, /1/"AU"
        	            	  | 1, NULL, 7, /1/"AU"/PrefixEnd, /1/"CA"
        	            	  | 1, NULL, 7, /1/"CA"/PrefixEnd, /1/"NZ"
        	            	  | 1, NULL, 7, /1/"NZ"/PrefixEnd, /1/"US"
        	            	  | 1, NULL, 7, /1/"US"/PrefixEnd, /2
        	            	  | 3, NULL, 7, /3, /3/"AU"
        	            	  | 3, NULL, 7, /3/"AU"/PrefixEnd, /3/"CA"
        	            	  | 3, NULL, 7, /3/"CA"/PrefixEnd, /3/"NZ"
        	            	  | 3, NULL, 7, /3/"NZ"/PrefixEnd, /3/"US"
        	            	  | 3, NULL, 7, /3/"US"/PrefixEnd, /4
        	            	  | 4, NULL, 7, /4, /4/"AU"
        	            	  | 4, NULL, 7, /4/"AU"/PrefixEnd, /4/"CA"
        	            	  | 4, NULL, 7, /4/"CA"/PrefixEnd, /4/"NZ"
        	            	  | 4, NULL, 7, /4/"NZ"/PrefixEnd, /4/"US"
        	            	  | 4, NULL, 7, /4/"US"/PrefixEnd, /5
        	            	  | 2, NULL, 8, /2, NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f25dcecd1fb55433d491511fa2e2d64f/logTestSubzonesRemovedByGCAfterIndexSwap1024276496
--- FAIL: TestSubzonesRemovedByGCAfterIndexSwap (32.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSubzonesRemovedByGCAfterIndexSwap.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49950

ccl/schemachangerccl.TestSubzonesRemovedByGCAfterIndexSwap [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19580516?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19580516?buildTab=artifacts#/) on master @ [93518dd0583a459870d670219c7e58abf4ad69de](https://github.com/cockroachdb/cockroach/commits/93518dd0583a459870d670219c7e58abf4ad69de):


```
        	            	  | 3, old_au, 5, /3/"AU", /3/"AU"/9131
        	            	  | 3, old_au, 5, /3/"NZ", /3/"NZ"/9131
        	            	  | 3, yung_au, 6, /3/"AU"/9131, /3/"AU"/PrefixEnd
        	            	  | 3, yung_au, 6, /3/"NZ"/9131, /3/"NZ"/PrefixEnd
        	            	  | 3, NULL, 7, /3, /3/"AU"
        	            	  | 3, NULL, 7, /3/"AU"/PrefixEnd, /3/"CA"
        	            	  | 3, NULL, 7, /3/"CA"/PrefixEnd, /3/"NZ"
        	            	  | 3, NULL, 7, /3/"NZ"/PrefixEnd, /3/"US"
        	            	  | 3, NULL, 7, /3/"US"/PrefixEnd, /4
        	            	  | 2, NULL, 8, /2, NULL
        	            	  |
        	            	  | got:
        	            	  | 1, north_america, 4, /1/"CA", NULL
        	            	  | 1, north_america, 4, /1/"US", NULL
        	            	  | 3, north_america, 4, /3/"CA", NULL
        	            	  | 3, north_america, 4, /3/"US", NULL
        	            	  | 4, north_america, 4, /4/"CA", NULL
        	            	  | 4, north_america, 4, /4/"US", NULL
        	            	  | 1, old_au, 5, /1/"AU", /1/"AU"/9131
        	            	  | 1, old_au, 5, /1/"NZ", /1/"NZ"/9131
        	            	  | 3, old_au, 5, /3/"AU", /3/"AU"/9131
        	            	  | 3, old_au, 5, /3/"NZ", /3/"NZ"/9131
        	            	  | 4, old_au, 5, /4/"AU", /4/"AU"/9131
        	            	  | 4, old_au, 5, /4/"NZ", /4/"NZ"/9131
        	            	  | 1, yung_au, 6, /1/"AU"/9131, /1/"AU"/PrefixEnd
        	            	  | 1, yung_au, 6, /1/"NZ"/9131, /1/"NZ"/PrefixEnd
        	            	  | 3, yung_au, 6, /3/"AU"/9131, /3/"AU"/PrefixEnd
        	            	  | 3, yung_au, 6, /3/"NZ"/9131, /3/"NZ"/PrefixEnd
        	            	  | 4, yung_au, 6, /4/"AU"/9131, /4/"AU"/PrefixEnd
        	            	  | 4, yung_au, 6, /4/"NZ"/9131, /4/"NZ"/PrefixEnd
        	            	  | 1, NULL, 7, /1, /1/"AU"
        	            	  | 1, NULL, 7, /1/"AU"/PrefixEnd, /1/"CA"
        	            	  | 1, NULL, 7, /1/"CA"/PrefixEnd, /1/"NZ"
        	            	  | 1, NULL, 7, /1/"NZ"/PrefixEnd, /1/"US"
        	            	  | 1, NULL, 7, /1/"US"/PrefixEnd, /2
        	            	  | 3, NULL, 7, /3, /3/"AU"
        	            	  | 3, NULL, 7, /3/"AU"/PrefixEnd, /3/"CA"
        	            	  | 3, NULL, 7, /3/"CA"/PrefixEnd, /3/"NZ"
        	            	  | 3, NULL, 7, /3/"NZ"/PrefixEnd, /3/"US"
        	            	  | 3, NULL, 7, /3/"US"/PrefixEnd, /4
        	            	  | 4, NULL, 7, /4, /4/"AU"
        	            	  | 4, NULL, 7, /4/"AU"/PrefixEnd, /4/"CA"
        	            	  | 4, NULL, 7, /4/"CA"/PrefixEnd, /4/"NZ"
        	            	  | 4, NULL, 7, /4/"NZ"/PrefixEnd, /4/"US"
        	            	  | 4, NULL, 7, /4/"US"/PrefixEnd, /5
        	            	  | 2, NULL, 8, /2, NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f25dcecd1fb55433d491511fa2e2d64f/logTestSubzonesRemovedByGCAfterIndexSwap1835227451
--- FAIL: TestSubzonesRemovedByGCAfterIndexSwap (32.27s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSubzonesRemovedByGCAfterIndexSwap.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/schemachangerccl/schemachanger_ccl_test.go
+++ b/pkg/ccl/schemachangerccl/schemachanger_ccl_test.go
@@ -9,7 +9,6 @@ import (
 	"context"
 	gosql "database/sql"
 	"testing"
-	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils"
@@ -248,7 +247,6 @@ CREATE TABLE person (
 	})
 
 	// Keep retrying until the old index and temporary index are removed by the GC job.
-	runner.SucceedsSoonDuration = 30 * time.Second
 	runner.CheckQueryResultsRetry(t, subzonesQuery, [][]string{
 		{"3", "north_america", "4", `/3/"CA"`, "NULL"},
 		{"3", "north_america", "4", `/3/"US"`, "NULL"},

